pkg/controller/datavolume-controller.go

```diff
@@ -2802,6 +2805,22 @@ func (r *DatavolumeReconciler) garbageCollect(dataVolume *cdiv1.DataVolume, pvc
 	return &reconcile.Result{}, nil
 }
 
+// Since PVC gets its DV ownerRefs which can be any CR, if BlockOwnerDeletion we allow GC only if we know CDI has RBAC
+// to update the owner finalizers (currently it's allowed only for VirtualMachines). BlockOwnerDeletions gets validated
+// with this admission plugin which is enabled in OpenShift, but disabled in our kubevirtci clusters:
+// https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
+func isGarbageCollectionAllowed(dv *cdiv1.DataVolume, log logr.Logger) bool {
+	for _, dvOwnerRef := range dv.OwnerReferences {
+		if dvOwnerRef.Kind != "VirtualMachine" &&
```

Comment:
I would suggest using the authorization APIs to check if you are able to update the finalizer for the parent object rather than checking for a specific type.  If you do this, then someone can make garbage collection work for any parent type so long as the provide the required rbac rules in the deployment.

Reply:
> provide the required rbac rules in the deployment

As in, they would provide the RBAC to the cdi controller service account?

Reply:
Thanks @aglitke, now using authorization API.

Reply:
Great idea @aglitke!